src/lib/y2network/dialogs/s390_device_activation.rb

```diff
@@ -100,6 +108,31 @@ def run
       def abort_handler
         Yast::Popup.ReallyAbort(true)
       end
+
+    private
+
+      def show_activation_error(message)
+        Yast2::Popup.show(
+          # TRANSLATORS: s390 group device activation error report. %s are the
+          # details of the error
+          _("An error occurred while activating the device.<br/><br/>" \
+            "<b>Details:<b/><br/>%s") % message,
+          richtext: true, headline: :error
+        )
+      end
+
+      def reader
+        @reader ||= Y2Network::Sysconfig::InterfacesReader.new
+      end
+
+      def config
+        Yast::Lan.yast_config
+      end
+
+      def add_interface(name)
+        interface = reader.interfaces.by_name(name)
+        config.interfaces << interface if interface
```

Comment:
this looks a bit strange for me. In which case it can fail to find interface?

Reply:
It should not fail, I could assign directly the reader.interfaces.by_name(name) interface.